actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java

```diff
@@ -139,6 +137,30 @@ private void createCourseBatch(Request actorMessage) {
     if (courseNotificationActive()) {
       batchOperationNotifier(courseBatch, null);
     }
+    if (CourseBatchUtil.doOperationInEkStepCourse(contentDetails, courseBatch, true)
+        .equalsIgnoreCase(JsonKey.SUCCESS)) {
+      courseBatch.setCountIncrementStatus(true);
+      try {
+        courseBatch.setCountIncrementDate(new SimpleDateFormat("yyyy-MM-dd").format(new Date()));
+        Map<String, Object> courseBatchMap =
+            new ObjectMapper().convertValue(courseBatch, Map.class);
+        Response response = courseBatchDao.update(courseBatchMap);
+
+        if (((String) response.get(JsonKey.RESPONSE)).equalsIgnoreCase(JsonKey.SUCCESS)) {
+          CourseBatchUtil.syncCourseBatchForeground(
+              (String) courseBatchMap.get(JsonKey.ID), courseBatchMap);
+        } else {
+          ProjectLogger.log(
+              "CourseBatchManagementActor:createCourseBatch: Course batch not synced to ES as response is not successful",
+              LoggerEnum.INFO.name());
+        }
+      } catch (Exception ex) {
+        ProjectLogger.log(
+            "CourseBatchManagementActor:createCourseBatch: Exception occurred with error message = "
+                + ex.getMessage(),
+            ex);
+      }
+    }
```

Comment:
break-down the function into smaller chunks - function has grown to too many lines.